includes/html/forms/application-update.inc.php

```diff
@@ -32,21 +34,24 @@
         $status = ['status' => 1, 'message' => 'Error with data'];
     } else {
         $status = ['status' => 1, 'message' => 'Database update failed'];
+        $app = Application::withTrashed()->firstOrNew(['device_id' => $device_id, 'app_type' => $app]);
         if ($_POST['state'] == 'true') {
-            $update = [
-                'device_id' => $device_id,
-                'app_type' => $app,
-                'app_status' => '',
-                'app_instance' => '',
-            ];
-            if (dbInsert($update, 'applications')) {
+            if ($app->trashed()) {
+                $app->restore();
+            }
+            if ($app->save()) {
                 log_event("Application enabled by user: $app", $device_id, 'application', 1);
                 $status = ['status' => 0, 'message' => 'Application enabled'];
+            } else {
+                $status = ['status' => 1, 'message' => 'Database update for enabling the application failed'];
             }
         } else {
-            if (dbDelete('applications', '`device_id`=? AND `app_type`=?', [$device_id, $app])) {
-                log_event("Application disabled by user: $app", $device_id, 'application', 3);
+            $app->delete();
+            if ($app->save()) {
+                log_event("Application disablaed by user: $app", $device_id, 'application', 3);
```

Comment:
Spelling here.  While we are making changes in this code, might as well change from log_event() to 
`Eventlog::log()`